python/pyarrow/array.pxi

```diff
@@ -2638,20 +2651,66 @@ cdef class ListViewArray(Array):
         Examples
         --------
         >>> import pyarrow as pa
-        >>> values = [[1, 2], [3, 4, None, 6]]
-        >>> offsets = [0, None, 2]
+        >>> values = [1, 2, None, 3, 4]
+        >>> offsets = [0, 0, 1]
         >>> sizes = [2, 0, 4]
         >>> array = pa.ListViewArray.from_arrays(offsets, sizes, values)
-        >>> array.offsets
+        >>> array.sizes
         <pyarrow.lib.Int32Array object at ...>
         [
+          2,
           0,
-          0,
-          2
+          4
         ]
         """
         return pyarrow_wrap_array((<CListViewArray*> self.ap).sizes())
 
+    def flatten(self, pool=None):
+        """
+        Unnest this ListViewArray by one level.
+
+        The returned Array is logically a concatenation of all the sub-lists
+        in this Array.
+
+        Note that this method is different from ``self.values`` in that
+        it takes care of the slicing offset as well as null elements backed
+        by non-empty sub-lists.
+
+        Parameters
+        ----------
+        pool : MemoryPool, optional
+
+        Returns
+        -------
+        result : Array
+
+        Examples
+        --------
+
+        >>> import pyarrow as pa
+        >>> values = [1, 2, 3, 4]
+        >>> offsets = [2, 1, 0]
+        >>> sizes = [2, 2, 2]
+        >>> array = pa.ListViewArray.from_arrays(offsets, sizes, values)
+        >>> array.flatten()
```

Comment:
Maybe here show both `array` and `array.values` outputs as well so it's easier to compare the different ones?